otherlibs/stdune/src/copyfile_stubs.c

```diff
@@ -0,0 +1,37 @@
+#include <caml/mlvalues.h>
+#include <caml/memory.h>
+
+#if defined(__APPLE__)
+#include <copyfile.h>
+#include <caml/threads.h>
+
+CAMLprim value stdune_copyfile(value v_from, value v_to) {
+  CAMLparam2(v_from, v_to);
+  copyfile_state_t state = copyfile_state_alloc();
+  const char *from = caml_stat_strdup(String_val(v_from));
+  const char *to = caml_stat_strdup(String_val(v_to));
+  caml_release_runtime_system();
+  copyfile(from, to, state, COPYFILE_CLONE);
```

Comment:
Thanks. I added error handling.

I don't think we need protect, because we don't need to worry about cleaning up resources like fd's.

Reply:
Since you've expressed interested, I've switched you to be the reviewer (I don't think Andrey uses macos anyway).

@nojbs can you have a look as well?